### ISSUE TITLE
Compiling dee-ubuntu in Arch Linux leads to "this 'while' clause does not guard..."

### ISSUE THREAD

**Issue by GraysonPeddie** [chenxiaolong/Unity-for-Arch#236]
I've been following the order of getting Unity compiled (https://github.com/chenxiaolong/Unity-for-Arch) in Arch Linux and got stuck with compiling dee-unity:

```
make  all-recursive
make[1]: Entering directory '/home/grayson/.src/Unity-for-Arch/dee-ubuntu/src/dee-1.2.7+15.04.20150304/build-python3'
Making all in build
make[2]: Entering directory '/home/grayson/.src/Unity-for-Arch/dee-ubuntu/src/dee-1.2.7+15.04.20150304/build-python3/build'
Making all in autotools
make[3]: Entering directory '/home/grayson/.src/Unity-for-Arch/dee-ubuntu/src/dee-1.2.7+15.04.20150304/build-python3/build/autotools'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/grayson/.src/Unity-for-Arch/dee-ubuntu/src/dee-1.2.7+15.04.20150304/build-python3/build/autotools'
make[3]: Entering directory '/home/grayson/.src/Unity-for-Arch/dee-ubuntu/src/dee-1.2.7+15.04.20150304/build-python3/build'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/home/grayson/.src/Unity-for-Arch/dee-ubuntu/src/dee-1.2.7+15.04.20150304/build-python3/build'
make[2]: Leaving directory '/home/grayson/.src/Unity-for-Arch/dee-ubuntu/src/dee-1.2.7+15.04.20150304/build-python3/build'
Making all in src
make[2]: Entering directory '/home/grayson/.src/Unity-for-Arch/dee-ubuntu/src/dee-1.2.7+15.04.20150304/build-python3/src'
make  all-am
make[3]: Entering directory '/home/grayson/.src/Unity-for-Arch/dee-ubuntu/src/dee-1.2.7+15.04.20150304/build-python3/src'
  CC       libdee_1_0_la-dee-serializable-model.lo
../../src/dee-serializable-model.c: In function ‘dee_serializable_model_get_position’:
../../src/dee-serializable-model.c:1329:3: error: this ‘while’ clause does not guard... [-Werror=misleading-indentation]
   while (!dee_model_is_last (self, iter) && iter != _iter)
   ^~~~~
../../src/dee-serializable-model.c:1331:5: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘while’
     pos++;
     ^~~
cc1: all warnings being treated as errors
make[3]: *** [Makefile:769: libdee_1_0_la-dee-serializable-model.lo] Error 1
make[3]: Leaving directory '/home/grayson/.src/Unity-for-Arch/dee-ubuntu/src/dee-1.2.7+15.04.20150304/build-python3/src'
make[2]: *** [Makefile:523: all] Error 2
make[2]: Leaving directory '/home/grayson/.src/Unity-for-Arch/dee-ubuntu/src/dee-1.2.7+15.04.20150304/build-python3/src'
make[1]: *** [Makefile:523: all-recursive] Error 1
make[1]: Leaving directory '/home/grayson/.src/Unity-for-Arch/dee-ubuntu/src/dee-1.2.7+15.04.20150304/build-python3'
make: *** [Makefile:429: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
```

Is there anything I can do to resolve the problem in my end?

**Comment by horsemanoffaith** [comment]
Take a look at my github page. I have an updated pkgbuild for dee-ubuntu that will correct this issue. There is a bug report filed for that issue. It seems to only affect Arch Linux users (probably because we're using a higher version of GCC).
https://github.com/horsemanoffaith/Unity-for-Arch

**Comment by smamontov** [comment]
@horsemanoffaith thank you, appreciate you help ;)

**Comment by smamontov** [comment]
@horsemanoffaith could you please create a PR here with this change?

**Comment by horsemanoffaith** [comment]
@smamontov  Not sure what you mean by PR.

**Comment by winneon** [comment]
@horsemanoffaith By PR, they mean create a pull request for this change. It would help anybody in the future that is interested in using this repository.

**Comment by chenxiaolong** [comment]
This is fixed by commit a2e03f226fdc5c8472e74215365beb9dd069af89
